### PR TITLE
feat(bundle): adds a self-executing dev bundle (SFX).

### PR DESCRIPTION
**Pull request by rkirov**

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -863,6 +863,23 @@ gulp.task('bundle.js.dev', ['build.js.dev'], function() {
...
+  var devBundleConfig = merge(true, bundleConfig);
+  devBundleConfig.paths =
+      merge(true, devBundleConfig.paths, {
+       '*': 'dist/js/dev/es6/*.es6'
+      });
+  return bundler.bundle(
+      devBundleConfig,
+      'angular2/angular2_sfx',
+      './dist/build/angular2.sfx.dev.js',
+      {
```
Comment (mprobst) [inline gulpfile.js]: put this on one line?

File: gulpfile.js [hunk 2]
```diff
@@ -875,17 +892,27 @@ gulp.task('bundle.js.min.deps', ['bundle.js.min'], function() {
...
       'angular2.dev.js')
       .pipe(insert.append('\nzone = zone.fork(Zone.longStackTraceZone);\n'))
       .pipe(gulp.dest('dist/bundle'));
 });
 
-gulp.task('bundle.js.deps', ['bundle.js.prod.deps', 'bundle.js.dev.deps', 'bundle.js.min.deps']);
+gulp.task('bundle.js.sfx.dev.deps', ['bundle.js.sfx.dev'], function() {
+  return bundler.modify(
+      jsDevDeps.concat(['dist/build/angular2.sfx.dev.js']),
+      'angular2.sfx.dev.js')
```
Comment (mprobst) [inline gulpfile.js]: nit: indentation

File: gulpfile.js [hunk 3]
```diff
@@ -863,6 +863,23 @@ gulp.task('bundle.js.dev', ['build.js.dev'], function() {
       });
 });
 
+// self-executing development build
```
Comment (mprobst) [inline gulpfile.js]: I think you need to explain what a self executing build is, at least to me that's not standard terminology.

File: gulpfile.js [hunk 4]
```diff
@@ -863,6 +863,23 @@ gulp.task('bundle.js.dev', ['build.js.dev'], function() {
...
+       '*': 'dist/js/dev/es6/*.es6'
+      });
+  return bundler.bundle(
+      devBundleConfig,
+      'angular2/angular2_sfx',
+      './dist/build/angular2.sfx.dev.js',
+      {
+        sourceMaps: true
+      }, 
+      true);  // self-executing bundle
```
Comment (mprobst) [inline gulpfile.js]: why isn't this on the outputConfig?

Comment (mprobst) [inline gulpfile.js]: I think the standard is "/* selfExecutingBundle */ true);"

File: gulpfile.js [hunk 5]
```diff
@@ -875,17 +892,27 @@ gulp.task('bundle.js.min.deps', ['bundle.js.min'], function() {
       .pipe(gulp.dest('dist/bundle'));
 });
 
+var jsDevDeps = ['node_modules/zone.js/zone.js', 'node_modules/zone.js/long-stack-trace-zone.js'];
```
Comment (mprobst) [inline gulpfile.js]: UPPER_CASE_CONSTANT?

Comment (rkirov) [inline gulpfile.js]: done.

File: gulpfile.js (hunk 4 above)
Comment (rkirov) [inline gulpfile.js]: the flag modifies whether we call `build` or `buildSfx`, why did the system-builder author decide to split those as separate methods outside config is beyond me.

moved comment.

File: gulpfile.js (hunk 3 above)
Comment (rkirov) [inline gulpfile.js]: added.

File: gulpfile.js (hunk 1 above)
Comment (rkirov) [inline gulpfile.js]: done.

File: gulpfile.js (hunk 2 above)
Comment (rkirov) [inline gulpfile.js]: replaced with `clang-format` output

**Comment by mprobst** [comment]
LGTM

**Pull request closed without merging**